core/xmake.lua

```diff
@@ -50,8 +50,8 @@ end
 
 -- for mode coverage
 if is_mode("coverage") then
-    add_cxflags("-coverage")
-    add_ldflags("-coverage")
+    add_cxflags("-coverage", "-fprofile-arcs", "-ftest-coverage")
+    add_ldflags("-coverage", "-fprofile-arcs", "-ftest-coverage")
```

Comment:
The old version gcov does not support the following structure generated by `xmake run`: 

```
./build/.objs/
└── testgcov
    └── src
        ├── main.c.d
        ├── main.c.gcda
        ├── main.c.gcno
        └── main.c.o
├── src
│   ├── main.c
```